sockeye/training.py

```diff
@@ -142,9 +151,21 @@ def sym_gen(seq_lens):
             # logits: (batch_size * target_seq_len, target_vocab_size)
             logits = self.output_layer(target_decoded)
 
-            loss_output = self.model_loss.get_loss(logits, labels)
-
-            return mx.sym.Group(loss_output), data_names, label_names
+            # 1) standard cross-entorpy loss
+            net_outputs = [self.model_loss.get_loss(logits, labels)]
+            # 2) length task losses
+            if self.length_task_loss is not None:
+                # predicted_length_ratios: (batch_size, 1)
+                predicted_length_ratios = self.length_ratio(source_encoded, source_encoded_length)
+                if isinstance(self.length_task_loss, loss.MSELoss):
+                    loss_symbol = self.length_task_loss.get_loss(predicted_length_ratios, length_ratio)
+                elif isinstance(self.length_task_loss, loss.PoissonLoss):
+                    # convert float predictions to integer lengths for the Poisson loss to be well defined
+                    predicted_reference_lengths = predicted_length_ratios * source_encoded_length.reshape((-1, 1))
+                    loss_symbol = self.length_task_loss.get_loss(predicted_reference_lengths, target_length)
+                # return both the loss symbol and the computed length_ratio to be used in metrics
+                net_outputs.extend([loss_symbol, mx.sym.BlockGrad(length_ratio, name=C.LENRATIO_NAME)])
```

Comment:
this should probably be `[loss_symbol, BlockGrad(predicted_length_ratios, name=LENRATIO_OUTPUT), BlockGrad(length_ratio, name=LENRATIO_LABEL)]`

Reply:
Thanks, now should return all three things in the Group (four with the softmax symbol)